src/main/java/org/openlmis/ao/reports/web/EmbeddedReportController.java

```diff
@@ -100,6 +101,10 @@ public EmbeddedReportDto createEmbeddedReport(@RequestBody EmbeddedReportDto dto
     permissionService.canViewEmbeddedReports();
     LOGGER.debug("Creating new embedded report");
 
+    if (dto.getName().isEmpty() || dto.getUrl().isEmpty() || dto.getCategory().isEmpty()) {
```

Comment:
lgtm. Could you also add a validation for `null` values?

Reply:
I refactored the validation to take null values into account, can you re-check it? @mgrochalskisoldevelo